Add ability to compile qml_box2d with system  Bod2D library

folibis opened the pull request:
request #95 suggests an idea to provide an ability to compile qml_box2d with system Box2D library. Sure it mostly related to *nix systems but anyway I think it's good idea. In my case I tested it in Debian and it gives 266К vs 422K of plugin size.

bjorn commented on box2d_lib.pri:
```diff
@@ -0,0 +1,58 @@
+INCLUDEPATH += $$PWD
+
+contains(DEFINES, BOX2D_SYSTEM) {
+    message("Compile qml_box2d with system Bod2D library")
```
I would say:

* "Compile" -> "Compiling"
* "qml_box2d" -> "qml-box2d"
* "Bod2D" -> "Box2D"

bjorn commented on box2d_lib.pri:
```diff
@@ -0,0 +1,58 @@
+        LIBS += "-lBox2D"
+
+} else {
+    message("Compile qml_box2d with builtin Bod2D library")
```
"builtin" -> "built-in"

bjorn commented on box2d.pro:
```diff
@@ -9,8 +9,13 @@ OBJECTS_DIR = .obj
 
-INCLUDEPATH += .
-include(Box2D/box2d.pri)
+# uncomment the line below to compile qml_box2d plugin with Box2D library, installed in OS
```
"qml_box2d" -> "qml-box2d"

bjorn commented on box2d.pro:
```diff
@@ -9,8 +9,13 @@ OBJECTS_DIR = .obj
-include(Box2D/box2d.pri)
+# uncomment the line below to compile qml_box2d plugin with Box2D library, installed in OS
+# warning: Box2D library must be already installed in system, for example in Debian it could be done with:
+# sudo apt-get install libbox2d2.3.0 libbox2d-dev
```
`libbox2d-dev` will pull in `libbox2d2.3.0` as dependency so it's not necessary to mention it explicitly.

bjorn commented on box2d.pro:
```diff
@@ -9,8 +9,13 @@ OBJECTS_DIR = .obj
+# warning: Box2D library must be already installed in system, for example in Debian it could be done with:
+# sudo apt-get install libbox2d2.3.0 libbox2d-dev
+
+#DEFINES += BOX2D_SYSTEM
```
Instead of uncommenting this, I think it is also possible to do:
```
qmake DEFINES+=BOX2D_SYSTEM
```
It would be good to mention this as well (when it is confirmed to work).

vranki commented:
$ dpkg -L libbox2d-dev |grep .pc
/usr/lib/x86_64-linux-gnu/pkgconfig/box2d.pc

box2d comes with a pkgconfig file. So could this be done easier with Qt's pkgconfig mechanism?

CONFIG += link_pkgconfig
PKGCONFIG += box2d

This would add the required include paths and libraries automatically to compiler command.

bjorn commented:
Well, Box2D doesn't come with a pkgconfig file. The Box2D package in Debian comes with a pkgconfig file. Anyway, it may still be preferable to use that mechanism.

folibis commented:
Thanks, @bjorn for the revision, english grammar is not my strength. :disappointed: I've updated the patch. I've also added pkgconfig support, as @vranki adviced.

bjorn commented on box2d.pro:
```diff
@@ -9,8 +9,15 @@ OBJECTS_DIR = .obj
+# Warning: Box2D library must be already installed in system, for example in Debian it could be done with:
+# sudo apt-get install libbox2d-dev
+
+DEFINES += BOX2D_SYSTEM
```
Well, this should not be enabled by default. :-)

bjorn commented on box2d_lib.pri:
```diff
@@ -0,0 +1,62 @@
+
+contains(DEFINES, BOX2D_SYSTEM) {
+
+    packagesExist(box2d) {
```
I didn't know about `packagesExist`, that's pretty neat with the fallback!

folibis commented:
Fixed.

bjorn commented:
Looks fine to me now, feel free to push to master.
